lib/attack.go

```diff
@@ -223,7 +223,61 @@ func Client(c *http.Client) func(*Attacker) {
 	return func(a *Attacker) { a.client = *c }
 }
 
-// A Rate of hits during an Attack.
+// A Rater defines the rate of hits during an Attack by
+// returning the duration an Attacker should wait until hiting
+// the next Target. When a negative value time is returned,
+// an Attacker will stop.
+type Rater interface {
+	Wait(began, now time.Time) time.Duration
```

Comment:
Function signature looks weird and counter-intuitive to me ... why do we need `began` and `now` parameters in order to be able to return `time.Duration` on each consequtive call ?

I think we leak implementation details here ...
What `Rater` basically is - an integer (time.Duration if we want to be more specific) sequence generator, independent of the concept of time.

Reply:
You are right, I haven't thought it through.

Reply:
> For testing purposes it's simple to have a separate internal implementation that takes a now parameter, as I did in #380.

Yeah, it was for testing purposes. I don't particularly mind using an API that makes me call time.Now() before, but I understand that's not everyone's cup of tea. So I'm cool with making it more ergonomic.